### PR TITLE
specify `docker run --rm` action when it comes to volume

### LINKED ISSUE #15455: docker run --rm documentation should indicate whether volumes are removed
It's not clear from any documentation that I can find whether `docker run --rm` is equivalent to doing a `docker rm -v` on the created container. Caveat: I usually can't find the mustard in the fridge because I'm looking everywhere *except* right in front of me.

### PR DESCRIPTION
**Pull request by dalanlan**
Fix #15455

### CODE REVIEW

**Comment by SvenDowideit** [comment]
ooo,really? nice - I was wondering about this today :)

LGTM

File: docs/reference/run.md [hunk 1]
```diff
@@ -463,6 +463,9 @@ the container exits**, you can add the `--rm` flag:
 
     --rm=false: Automatically remove the container when it exits (incompatible with -d)
 
+> Note that when you set `--rm` flag, Docker removes the volumes associated
```
Comment (thaJeztah) [inline docs/reference/run.md]: Can you change this to our standard "note" format? Something like;

```Markdown
> **Note**: When you set the `--rm` flag, Docker also removes the volumes associated with
> the container when the container is removed. This is similar to running `docker rm -v my-container`.

Comment (dalanlan) [inline docs/reference/run.md]: Addressed. Thx.

Comment (thaJeztah) [inline docs/reference/run.md]: Hm, reading back, think I should've suggested "if you set". Guess it's good enough for now :relaxed:

**Comment by thaJeztah** [comment]
LGTM, thanks!

**Comment by posita** [comment]
@dalanlan, thanks for getting to this before I could! :+1:

**Pull request merged**